src/lib/utils/hooks/SynapseAPI/useUserBundle.ts

```diff
@@ -1,22 +1,48 @@
-import { useQuery, UseQueryOptions } from 'react-query'
+import {
+  QueryObserverOptions,
+  useQuery,
+  useQueryClient,
+  UseQueryOptions,
+} from 'react-query'
 import {
   getUserProfileWithProfilePic,
   UserProfileAndImg,
 } from '../../functions/getUserData'
 import { SynapseClientError } from '../../SynapseClient'
 import { useSynapseContext } from '../../SynapseContext'
+import { UserProfile } from '../../synapseTypes'
 
 export function useGetUserProfileWithProfilePic(
   principalId: string,
-  options?: UseQueryOptions<
-    UserProfileAndImg,
-    SynapseClientError,
-    UserProfileAndImg
-  >,
+  options?: UseQueryOptions<UserProfileAndImg, SynapseClientError>,
 ) {
   const { accessToken } = useSynapseContext()
+  const queryClient = useQueryClient()
+  const queryKey = [accessToken, 'user', principalId, 'profile', 'withPic']
+
+  // We store the profile in a session storage cache used by SWC
+  const sessionStorageCacheKey = `${principalId}_USER_PROFILE`
+  const cachedValue = sessionStorage.getItem(sessionStorageCacheKey)
+
+  const defaultOptions: QueryObserverOptions<
+    UserProfileAndImg,
+    SynapseClientError
+  > = {
+    // Use the sessionStorage cache to pre-populate profile data before the request succeeds.
+    placeholderData: cachedValue
+      ? { userProfile: JSON.parse(cachedValue) as UserProfile }
+      : undefined,
```

Comment:
just a note, react-query also has `initialData`, but that gets cached, while `placeholderData` does not